model/modeldecoder/v2/decoder.go

```diff
@@ -67,6 +68,11 @@ var (
 	}
 )
 
+var (
+	patternWithSingleSlash   = regexp.MustCompile(`^([a-z]+)/(\w+)$`)
+	patternWithOnlyAlphabets = regexp.MustCompile(`^[a-z]+$`)
```

Comment:
It should be possible to use a single regex with a non-capturing group for the slash (`/`):

```suggestion
	patternServiceTargetRe = regexp.MustCompile(`^([a-z]+)(?:/)?(\w+)?$`)
```

Reply:
Hmm, yes this should be possible but we will have to put `\w+` also in the first non-capturing group along with `/` otherwise abcd2343 will also match. Let me think about it.

Reply:
Using `^([a-z]+)(?:/(\w+))?$` as regex is one possible solution for utilising single regex ([rough draft on go playground](https://go.dev/play/p/6Hkm9tSg1zA)). Readability wise I am a tiny bit more inclined towards using separate regexes. WDYT?

Reply:
I don't have a preference between `^([a-z]+)(?:/)?(\w+)?$` and `^([a-z]+)(?:/(\w+))?$` since from what I can see they're identical.

On readability, I don't think having two regexes makes it more readable, since there's more logic to handle on the `if/else` and I prefer to read a single regex that has a nice comment on it (explaining the different groups and what they intend to capture) rather than having an `if/else/else if` combination. Happy to have other members of the team chime in.

I also think that `w` and `[a-z]` are exactly the same so I'm not too worried bout which one to use either.

Reply:
> I don't have a preference between ^([a-z]+)(?:/)?(\w+)?$ and ^([a-z]+)(?:/(\w+))?$ since from what I can see they're identical.

With the former, as per my understanding, it is not possible to distinguish between `postgres/123` and `postgres123`. Both of these will lead to submatch array of length 3. On the other hand. the latter expression will not match for `postgres123`

Reply:
> I also think that w and [a-z] are exactly the same so I'm not too worried bout which one to use either.

`w` should be equivalent to `[a-zA-Z0-9_]`

Reply:
I generally find regexes hard to understand and am a fan of simplifying them for readability. But no strong feelings, whichever one of the two discussed solutions you move forward are fine with me. 
+1 on adding a comment what the regex is intended to capture.

Reply:
On thinking about it, I still don't have a strong inclination towards either of the two approaches. The approach involving a single regex doesn't lead to a very complicated regex or logic so I decided to update the PR with the single regex approach.